script.module.dropbox/resources/utils.py

```diff
@@ -0,0 +1,58 @@
+"""
+This file contains additional utility functions
+"""
+import xbmc, xbmcgui, xbmcvfs, xbmcaddon
+import os
+import sys
+
+__addon_id__= u'script.module.dropbox'
+__Addon = xbmcaddon.Addon(__addon_id__)
+
+
+def data_dir():
+    """"get user data directory of this addon. 
+    according to http://wiki.xbmc.org/index.php?title=Add-on_Rules#Requirements_for_scripts_and_plugins
+    """
+    __datapath__ = xbmc.translatePath( __Addon.getAddonInfo('profile') ).decode('utf-8')
+    if not xbmcvfs.exists(__datapath__):
+        xbmcvfs.mkdir(__datapath__)
+    return __datapath__
+
+def addon_dir():
+    """"get source directory of this addon.
+    according to http://wiki.xbmc.org/index.php?title=Add-on_Rules#Requirements_for_scripts_and_plugins
+    """
+    return __Addon.getAddonInfo('path').decode('utf-8')
+
+def encode(string):
+    return string.encode('UTF-8','replace')
+
+def decode(string):
+    return string.decode('UTF-8') 
+
+def log(message,loglevel=xbmc.LOGNOTICE):
```

Comment:
Can you please set a minimum log level of LOGDEBUG.

Reply:
Hi -- I've made all the requested (mandatory) changes.

Thanks in advance.

Reply:
minimum log level of LOGDEBUG.